### ISSUE TITLE
TypeError: DatetimeIndex cannot perform the operation sum

### ISSUE THREAD

**Issue by VsmashJ** [fbdesignpro/sweetviz#19]
TypeError                                 Traceback (most recent call last)
 in 
      1 #analyzing the dataset
----> 2 report = sv.analyze(channelmemberhistory)
      3 #display the report
      4 report.show_html('MemberHistory.html')

~/.local/lib/python3.6/site-packages/sweetviz/sv_public.py in analyze(source, target_feat, feat_cfg, pairwise_analysis)
     11             pairwise_analysis: str = 'auto'):
     12     report = sweetviz.DataframeReport(source, target_feat, None,
---> 13                                       pairwise_analysis, feat_cfg)
     14     return report
     15 

~/.local/lib/python3.6/site-packages/sweetviz/dataframe_report.py in __init__(self, source, target_feature_name, compare, pairwise_analysis, fc)
    214             # start = time.perf_counter()
    215             self.progress_bar.set_description(':' + f.source.name + '')
--> 216             self._features[f.source.name] = sa.analyze_feature_to_dictionary(f)
    217             self.progress_bar.update(1)
    218             # print(f"DONE FEATURE------> {f.source.name}"

~/.local/lib/python3.6/site-packages/sweetviz/series_analyzer.py in analyze_feature_to_dictionary(to_process)
     90 
     91     # Establish base stats
---> 92     add_series_base_stats_to_dict(to_process.source, to_process.source_counts, returned_feature_dict)
     93     if to_process.compare is not None:
     94         add_series_base_stats_to_dict(to_process.compare, to_process.compare_counts, compare_dict)

~/.local/lib/python3.6/site-packages/sweetviz/series_analyzer.py in add_series_base_stats_to_dict(series, counts, updated_dict)
     42     base_stats = updated_dict["base_stats"]
     43     num_total = counts["num_rows_total"]
---> 44     num_zeros = series[series == 0].sum()
     45     non_nan = counts["num_rows_with_data"]
     46     base_stats["total_rows"] = num_total

~/.local/lib/python3.6/site-packages/pandas/core/generic.py in stat_func(self, axis, skipna, level, numeric_only, min_count, **kwargs)
  11180             skipna=skipna,
  11181             numeric_only=numeric_only,
> 11182             min_count=min_count,
  11183         )
  11184 

~/.local/lib/python3.6/site-packages/pandas/core/series.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   3901             numeric_only=numeric_only,
   3902             filter_type=filter_type,
-> 3903             **kwds,
   3904         )
   3905 

~/.local/lib/python3.6/site-packages/pandas/core/base.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   1058         if func is None:
   1059             raise TypeError(
-> 1060                 f"{type(self).__name__} cannot perform the operation {name}"
   1061             )
   1062         return func(skipna=skipna, **kwds)

**Comment by gouthamreddy09** [comment]
@VsmashJ are using .xlsx or .csv?

**Comment by VsmashJ** [comment]
Pandas Data Frame from Mysql and Last field transformed of Unix Epoch Timestamp converted to datetime using to_datatime() pandas

**Comment by rd5802** [comment]
Error with  datetime64[ns] field. Pandas 1.0.3.

```
import pandas as pd
import sweetviz as sv

df = pd.read_sas('file.sas7bdat', encoding='latin2')
report = sv.analyze(df)
```
```

:date_field:                   |█▌                   | [  8%]   00:00  -> (00:09 left)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-36ccb6f5383f> in <module>
      1 df = pd.read_sas('idl_linear_reg_mod_cala.sas7bdat', encoding='latin2')
----> 2 report = sv.analyze(df)

~/eksperyment/lib64/python3.6/site-packages/sweetviz/sv_public.py in analyze(source, target_feat, feat_cfg, pairwise_analysis)
     11             pairwise_analysis: str = 'auto'):
     12     report = sweetviz.DataframeReport(source, target_feat, None,
---> 13                                       pairwise_analysis, feat_cfg)
     14     return report
     15 

~/eksperyment/lib64/python3.6/site-packages/sweetviz/dataframe_report.py in __init__(self, source, target_feature_name, compare, pairwise_analysis, fc)
    214             # start = time.perf_counter()
    215             self.progress_bar.set_description(':' + f.source.name + '')
--> 216             self._features[f.source.name] = sa.analyze_feature_to_dictionary(f)
    217             self.progress_bar.update(1)
    218             # print(f"DONE FEATURE------> {f.source.name}"

~/eksperyment/lib64/python3.6/site-packages/sweetviz/series_analyzer.py in analyze_feature_to_dictionary(to_process)
     90 
     91     # Establish base stats
---> 92     add_series_base_stats_to_dict(to_process.source, to_process.source_counts, returned_feature_dict)
     93     if to_process.compare is not None:
     94         add_series_base_stats_to_dict(to_process.compare, to_process.compare_counts, compare_dict)

~/eksperyment/lib64/python3.6/site-packages/sweetviz/series_analyzer.py in add_series_base_stats_to_dict(series, counts, updated_dict)
     42     base_stats = updated_dict["base_stats"]
     43     num_total = counts["num_rows_total"]
---> 44     num_zeros = series[series == 0].sum()
     45     non_nan = counts["num_rows_with_data"]
     46     base_stats["total_rows"] = num_total

~/eksperyment/lib64/python3.6/site-packages/pandas/core/generic.py in stat_func(self, axis, skipna, level, numeric_only, min_count, **kwargs)
  11180             skipna=skipna,
  11181             numeric_only=numeric_only,
> 11182             min_count=min_count,
  11183         )
  11184 

~/eksperyment/lib64/python3.6/site-packages/pandas/core/series.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   3901             numeric_only=numeric_only,
   3902             filter_type=filter_type,
-> 3903             **kwds,
   3904         )
   3905 

~/eksperyment/lib64/python3.6/site-packages/pandas/core/base.py in _reduce(self, op, name, axis, skipna, numeric_only, filter_type, **kwds)
   1058         if func is None:
   1059             raise TypeError(
-> 1060                 f"{type(self).__name__} cannot perform the operation {name}"
   1061             )
   1062         return func(skipna=skipna, **kwds)

TypeError: DatetimeIndex cannot perform the operation sum
```

**Comment by kais-viz** [comment]
Facing a similar issue on Kaggle. 
Current doing an analysis on the [telco customer churn](https://www.kaggle.com/blastchar/telco-customer-churn) dataset. Initially on the raw data it works well and the report is generated, but once I change the datatype of some attributes to categorical, it crashes with a similar error to the above.

`
import sweetviz as sv
import pandas as pd

data = pd.read_csv(r"../input/telco-customer-churn/WA_Fn-UseC_-Telco-Customer-Churn.csv")
data["gender"] = data["gender"].astype('category')
my_report = sv.analyze(data)`

`
TypeError: Categorical cannot perform the operation sum
`

**Comment by kais-viz** [comment]
Facing a similar issue on Kaggle. 
Current doing an analysis on the [telco customer churn](https://www.kaggle.com/blastchar/telco-customer-churn) dataset. Initially on the raw data it works well and the report is generated, but once I change the datatype of some attributes to categorical, it crashes with a similar error to the above.

```
import sweetviz as sv
import pandas as pd

data = pd.read_csv(r"../input/telco-customer-churn/WA_Fn-UseC_-Telco-Customer-Churn.csv")
data["gender"] = data["gender"].astype('category')
my_report = sv.analyze(data)
```

`
TypeError: Categorical cannot perform the operation sum
`

**Comment by fbdesignpro** [comment]
Should be fixed with the latest changes! If you are using the repository, grabbing the latest will fix this. Otherwise a `pip` build will be done soon that contains fixes for this. I will close when it is out.

**Comment by fbdesignpro** [comment]
Closing, issue should be resolved in 1.0a8!
